### PR TITLE
[Feature][ci] Rerun failed step for the backend Github action

### LINKED ISSUE #1257: [Feature][ci] Rerun failed step for the backend Github action
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Sometimes, the `Build and Package` step in the `backend` Github Action may be pending, the reason may be that the Github Action is unstable due to the network. So, we should add a timeout option for the step and rerun the failed step for it to improve the sucess rate. If it is working, we may need generialize this to other Github Action jobs.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by asdf2014**
## Purpose of this pull request

Fixes #1257

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Review by CalvinKirs — COMMENTED** [review]
I think good, but it used doesn't look like it's allowed by ASF, can you help me reconfirm? @kezhenxu94

**Comment by asdf2014** [comment]
May the author can tell us more, cc @nick-invision

**Comment by CalvinKirs** [comment]
> May the author can tell us more, cc @nick-invision

you can refer this https://infra.apache.org/github-actions-secrets.html

**Comment by asdf2014** [comment]
Hi @CalvinKirs, thanks for your comments, it seems that I don't have permission to access the secrets of this project :sweat_smile:

image

**Comment by kezhenxu94** [comment]
> May the author can tell us more, cc @nick-invision

No need to mention people unless you are very certain otherwise it's kind of disturbing, according to ASF policy, third-party GitHub actions can not be executed in ASF repos unless they are allowlisted.

**Comment by kezhenxu94** [comment]
To retry a command (`xxx`), you can simply use something like `xxx || xxx`

**Comment by asdf2014** [comment]
Hi @kezhenxu94, thanks for your comment, in this case the job is on pending status, so we still need timeout the step and rerun it, so the `xxx || xxx` solution may not be enough.

**Comment by kezhenxu94** [comment]
> To retry a command (`xxx`), you can simply use something like `xxx || xxx`

If you insist using that GitHub Action, you need to add it as sub module or request the infra team to add it into allowlist

**Comment by kezhenxu94** [comment]
> Hi @kezhenxu94, thanks for your comment, in this case the job is on pending status, so we still need timeout the step and rerun it, so the `xxx || xxx` solution may not be enough.

I rarely see maven command itself can get stuck and timeout, can you give me an example case?

**Comment by asdf2014** [comment]
@kezhenxu94 FYI, https://github.com/apache/incubator-seatunnel/issues/1265

**Comment by asdf2014** [comment]
Hi @kezhenxu94, thanks for your suggestion, I just raised up a ticket to the Infra team, let's put eyes on it, FYI, https://issues.apache.org/jira/browse/INFRA-22891

**Comment by kezhenxu94** [comment]
> @kezhenxu94 FYI, https://github.com/apache/incubator-seatunnel/issues/1265

From the issue description,

> which is not necessary, maybe we can skip this operation.

Why do you bother to add this retry action instead of resolving that root problem? You are adding a new thing `C` to work around a problem `B`, which should be addressed first.

**Comment by asdf2014** [comment]
@kezhenxu94 Indeed, this is a work around solution, if my another solution https://github.com/apache/incubator-seatunnel/pull/1266 is working, I may close this one in future.

File: .github/workflows/backend.yml [hunk 1]
```diff
@@ -62,15 +62,20 @@ jobs:
...
-                 -Dmaven.test.skip=true \
-                 -Dcheckstyle.skip=true \
-                 -Dhttp.keepAlive=false \
-                 -Dmaven.wagon.http.pool=false \
-                 -Dmaven.wagon.http.retryHandler.count=3 \
-                 -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 \
-                 --no-snapshot-updates
+        uses: nick-invision/retry@v2
+        with:
+          timeout_minutes: 10
```
Comment (xleoken) [inline .github/workflows/backend.yml:67]: hi @asdf2014, `timeout_minutes: 10` may is not enough, seems most actions always beyond the limits.

**Comment by kezhenxu94** [comment]
Closing in favor of https://github.com/apache/incubator-seatunnel/pull/1266

File: .github/workflows/backend.yml (hunk 1 above)
Comment (asdf2014) [reply .github/workflows/backend.yml:67]: @leo65535 It is enough for this step

**Pull request closed without merging**